src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate.cs

```diff
@@ -316,12 +316,30 @@ public virtual byte[] GetCertHash()
             return GetRawCertHash().CloneByteArray();
         }
 
+        public virtual byte[] GetCertHash(HashAlgorithmName hashAlgorithm)
+        {
+            ThrowIfInvalid();
+
+            using (IncrementalHash hasher = IncrementalHash.CreateHash(hashAlgorithm))
```

Comment:
Why IncrementalHash? Wouldn't HashAlgorithm.ComputeHash be at least as efficient?

Reply:
Yep, but we don't have a thing that takes HashAlgorithmName and produces HashAlgorithm.  IncrementalHash already takes HashAlgorithmName as an input.